File: combat/src/classicthunder/combat/character/CharacterActor.java
package classicthunder.combat.character;

import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.g2d.BitmapFont;
import com.badlogic.gdx.graphics.g2d.Sprite;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.math.Vector2;

public class CharacterActor {

    private final Sprite sprite;
    private final CharacterStats characterStats;

    private final BitmapFont font;

    public CharacterActor(CharacterType characterType, Texture texture, Vector2 location) {

        Vector2 size = new Vector2(texture.getWidth(), texture.getHeight());

        sprite = new Sprite(texture);

        sprite.setOrigin(size.x / 2f, 0);
        sprite.setOriginBasedPosition(location.x, location.y);

        characterStats = new CharacterStats(characterType, 20);

        font = new BitmapFont();
    }

    public boolean ContainsMouse(Vector2 mouse) {

        return sprite.getBoundingRectangle().contains(mouse);
    }

    public CharacterStats GetStats() {

        return characterStats;
    }

    public void Draw(SpriteBatch batch) {

        sprite.draw(batch);

        font.draw(batch, "HP = " + characterStats.GetHealth(),
                sprite.getX(), sprite.getY() + sprite.getHeight());
        font.draw(batch, "Block = " + characterStats.GetBlock(),
                sprite.getX(), sprite.getY() + sprite.getHeight() + 14);

    }
}

File: combat/src/classicthunder/combat/character/CharacterActor.kt
package classicthunder.combat.character

import classicthunder.actor.SpriteActor
import classicthunder.character.NPC
import com.badlogic.gdx.graphics.Texture
import com.badlogic.gdx.graphics.g2d.BitmapFont
import com.badlogic.gdx.graphics.g2d.SpriteBatch
import com.badlogic.gdx.math.Vector2

internal open class CharacterActor(
    internal open val character: NPC,
    texture: Texture,
    location: Vector2
) : SpriteActor(
    texture,
    location,
    origin = Vector2(texture.width / 2f, 0f),
    size = Vector2(texture.width.toFloat(), texture.height.toFloat())
) {
    private val font: BitmapFont = BitmapFont()

    override fun draw(batch: SpriteBatch) {

        super.draw(batch)

        font.draw(batch, "HP = " + character.characterStats.GetHealth(),
            location.x, location.y)
        font.draw(batch, "Block = " + character.characterStats.GetBlock(),
            location.x, location.y - 14)
    }
}